migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs

```diff
@@ -288,10 +297,44 @@ impl<'schema> SqlSchemaDiffer<'schema> {
             .map(|pk| TableChange::AddPrimaryKey {
                 columns: pk.columns.clone(),
             })
+            .or_else(|| {
```

Comment:
OK, here they are what I was talking about with @tomhoule.

If we must drop and re-create the column on SQL Server, we have to drop the primary key constraint before dropping the column, then re-create it after creating the new column. Maybe we should do some flavor magic here?